osu.Game/Screens/Edit/Compose/Components/Timeline/TimelineBlueprintContainer.cs

```diff
@@ -193,16 +198,42 @@ bool shouldBeSelected(HitObject hitObject)
 
         private void handleScrollViaDrag()
         {
+            // The amount of time dragging before we reach maximum drag speed.
+            const float time_ramp_multiplier = 5000;
+
+            // A maximum drag speed to ensure things don't get out of hand.
+            const float max_velocity = 10;
+
             if (timeline == null) return;
 
-            var timelineQuad = timeline.ScreenSpaceDrawQuad;
-            float mouseX = InputManager.CurrentState.Mouse.Position.X;
+            var mousePos = timeline.ToLocalSpace(InputManager.CurrentState.Mouse.Position);
+
+            // for better UX do not require the user to drag all the way to the edge and beyond to initiate a drag-scroll.
+            // this is especially important in scenarios like fullscreen, where mouse confine will usually be on
+            // and the user physically *won't be able to* drag beyond the edge of the timeline
+            // (since its left edge is co-incident with the window edge).
+            const float scroll_tolerance = 40;
+
+            float leftBound = timeline.BoundingBox.TopLeft.X + scroll_tolerance;
+            float rightBound = timeline.BoundingBox.TopRight.X - scroll_tolerance;
+
+            float amount = 0;
+
+            if (mousePos.X > rightBound)
+                amount = mousePos.X - rightBound;
+            else if (mousePos.X < leftBound)
+                amount = mousePos.X - leftBound;
+
+            if (amount == 0)
+            {
+                dragTimeAccumulated = 0;
+                return;
+            }
+
+            amount = Math.Sign(amount) * Math.Min(max_velocity, Math.Abs(MathF.Pow(amount, 2) / (MathF.Pow(scroll_tolerance, 2))));
```

Comment:
I can see what you were going for with this and I was considering applying something similar before stopping myself. That said, I'd probably reshuffle this slightly to:

```suggestion
            amount = Math.Sign(amount) * Math.Min(max_velocity, MathF.Pow(Math.Clamp(Math.Abs(amount), 0, scroll_tolerance), 2));
```

My rationale would be that even with these latest changes scrolling to the left feels noticeably more sluggish than scrolling to the right, and clamping `amount` to the `scroll_tolerance` maximum should in theory alleviate it.